### PR TITLE
Showing up Site or folder field in contentlet selector

### LINKED ISSUE #27563: Site or Folder field does not show on the relate content window
### Parent Issue

https://github.com/dotCMS/core/issues/24194
https://github.com/dotCMS/core/issues/24290

### Problem Statement

Site or folder field does not show up on the relate content window. This is believed to stem from when we removed the 'User searchable' option was removed from the Site or Folder field, and instead it was added to Search automatically when the field was added to a content type. This was missed for the relate content window.

### Steps to Reproduce

1. From a demo starter, look at a Blog contentlet.
2. Try to relate a new author, you will see the Site or Folder available for search
3. Go To Content Types and select the Blog Author Content Type
4. Remove the Site or Folder Field.
5. Add the Site or Folder Field.
6. Try to relate a new author again, you will no longer see the Site or Folder field available

### Acceptance Criteria

You should be able to relate to content from other sites when the related content type has a site or folder field.

Let's do this like in the Content Search portlet, which always shows the host or folder field regardless if is user searchable or not.

### dotCMS Version

23.10.24, latest

### Proposed Objective

User Experience

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.zendesk.com/agent/tickets/115144

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by AndreyDotcms**
- Exclude the site folder fields from appearing as part of the content fields.
- Always append the site folder dropdown to the search bar

https://github.com/dotCMS/core/assets/127987858/368ba1b3-c8ce-41ec-a119-15c8278302ae

image
image

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/webapp/html/js/dotcms/dijit/form/ContentSelector.js [hunk 1]
```diff
@@ -305,22 +305,45 @@ dojo.declare(
             }
         },
 
+        _getSiteFolderFieldDefaultHTML: function (){
+            const defaultSiteFolderField = {
+                "fieldContentlet": "system_field",
+                "fieldFieldType": "host or folder",
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/webapp/html/js/dotcms/dijit/form/ContentSelector.js:311]: shouldn't this fieldFieldType be "host_or_folder" ?  I mean blanks in a type ?

File: dotCMS/src/main/webapp/html/js/dotcms/dijit/form/ContentSelector.js [hunk 2]
```diff
@@ -305,22 +305,45 @@ dojo.declare(
...
+
         _fillFields: function (data) {
             this.currentStructureFields = data;
             this.search_fields_table.innerHTML = '';
+            this.site_folder_field_pop.innerHTML = '';
             var htmlstr = "<dl class='vertical'>";
             for (var i = 0; i < data.length; i++) {
                 var type = data[i]['fieldFieldType'];
-                if (type == 'category' || type == 'hidden') {
+                if (type == 'category' || type == 'hidden' || type == "host or folder") {
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/webapp/html/js/dotcms/dijit/form/ContentSelector.js:332]: same here I would change type to  "host_or_folder"

**Review by fabrizzio-dotCMS — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/webapp/html/js/dotcms/dijit/form/ContentSelector.js (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/webapp/html/js/dotcms/dijit/form/ContentSelector.js:311]: Looks like this value comes from this const
`HOST_OR_FOLDER("host or folder")`
approving

**Review by fabrizzio-dotCMS — APPROVED** [review]

File: dotCMS/src/main/webapp/html/js/dotcms/dijit/form/ContentSelector.js (hunk 1 above)
Comment (AndreyDotcms) [reply dotCMS/src/main/webapp/html/js/dotcms/dijit/form/ContentSelector.js]: `Field.FieldType.HOST_OR_FOLDER.toString()` can only be accessed in JSP or Java files

File: dotCMS/src/main/webapp/html/js/dotcms/dijit/form/ContentSelector.js (hunk 2 above)
Comment (AndreyDotcms) [reply dotCMS/src/main/webapp/html/js/dotcms/dijit/form/ContentSelector.js]: `Field.FieldType.HOST_OR_FOLDER.toString()` can only be accessed in JSP or Java files

**Pull request merged**